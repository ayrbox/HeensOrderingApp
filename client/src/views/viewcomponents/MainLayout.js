import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import Footer from './Footer';
import Header from '../../containers/Header';

import styles from './mainLayoutStyles';

const MainLayout = ({ children, classes }) => (
  <Fragment>
    <Header title="Heens" />
    <main className={classes.wrapper}>
      <div className={classes.container}>
        {children}
      </div>
    </main>
    <Footer data-testid="footer" />
  </Fragment>
);

MainLayout.propTypes = {
  children: PropTypes.node.isRequired,
  classes: PropTypes.shape().isRequired,
};

export default withStyles(styles)(MainLayout);
