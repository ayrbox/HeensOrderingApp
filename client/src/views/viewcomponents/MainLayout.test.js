import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history'
import { render, cleanup } from 'react-testing-library'
import 'jest-dom/extend-expect';

import MainLayout from './Footer';

function renderWithRouter(
  ui, {
    route = '/',
    history = createMemoryHistory({initialEntries: [route]})
  } = {},
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history,
  }
}


describe('Main Layout', () => {
  afterEach(cleanup);

  it('MainLayout', () => {
    const mainLayout = renderWithRouter(<MainLayout />);
    expect(mainLayout).toMatchSnapshot();
  });
});
