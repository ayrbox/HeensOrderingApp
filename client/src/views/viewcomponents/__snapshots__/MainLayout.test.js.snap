// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout MainLayout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="Footer-footerWrapper-1"
      >
        <div
          class="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
        >
          <div
            class="MuiGrid-item-3"
          >
            <p
              data-testid="copyright-text"
            >
              ©
               2019 MIT License
            </p>
          </div>
          <div
            class="MuiGrid-item-3"
          >
            <p
              class="float-right"
            >
              <a
                data-testid="back-to-top"
                href="/"
              >
                Back to top
              </a>
            </p>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="Footer-footerWrapper-1"
    >
      <div
        class="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
      >
        <div
          class="MuiGrid-item-3"
        >
          <p
            data-testid="copyright-text"
          >
            ©
             2019 MIT License
          </p>
        </div>
        <div
          class="MuiGrid-item-3"
        >
          <p
            class="float-right"
          >
            <a
              data-testid="back-to-top"
              href="/"
            >
              Back to top
            </a>
          </p>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "history": Object {
    "action": "POP",
    "block": [Function],
    "canGo": [Function],
    "createHref": [Function],
    "entries": Array [
      Object {
        "hash": "",
        "key": "9ahxvd",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
    ],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "index": 0,
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "key": "9ahxvd",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
